apps/dashboard/src/components/contract-functions/contract-function.tsx

```diff
@@ -456,13 +467,24 @@ const FunctionsOrEventsListItem: React.FC<FunctionsOrEventsListItemProps> = ({
         }
         onClick={() => {
           setSelectedFunction(fn);
-          const { name } = fn;
           const path = router.asPath.split("?")[0];
           // Only apply to the Explorer page
           if (path.endsWith("/explorer")) {
-            router.push({ pathname: path, query: { name } }, undefined, {
-              shallow: true,
-            });
+            // Check if there are more than 1 function with the same name
+            const isOverloadFn =
+              (allFunctions || []).filter((o) => o.name === fn.name).length > 1;
+            const url = new URL(window.location.href);
+            url.searchParams.set("name", fn.name);
+            // In case there are more than 1 fn with the same name, distinguish them by their inputs
+            if (isOverloadFn) {
+              url.searchParams.set(
+                "inputs",
+                fn.inputs.map((i) => i.name).join(","),
```

Comment:
`name` is not guaranteed to exist, probably need to use the type or something here, maybe just the human readable abi or the fn selector?

Reply:
yes switching to fn signature - I can see that you have migrated this part to v5!